af/shovel/centrifugation.py

```diff
@@ -326,163 +329,491 @@ def __flush_stubborn(self):
     def close(self):
         self.flush()
         self.__buf.close()
+    @property
+    def closed(self):
+        return self.__buf.closed
 
-def get_checked_code_ver(pgconn, autoclaved_index, bucket):
-    # Returns:
-    #   None -- the bucket was never loaded
-    #   int  -- code version stamped in `autoclaved` table for these files
-    #
-    # Check that postgresql and autoclaved/${bucket}/index.json.gz ...
-    # - have same list of data files, names and SHA1 are equal
-    # - have same count of reports (type: report) and measurements (type: datum)
-    #
-    # Number of distinct frames in PG and {type: frame} may differ as there are frames without measurements:
-    # $ zcat 2017-03-21/index.json.gz | awk '/"type": ?"frame"/ {a = 0; s = ""} {s = s $0} /"type": ?"datum"/ { a +=1 } (/"type": ?"\/frame"/ && a == 0) {print s; s = ""}' | shuf -n 5
-    # {"file_off": 14628392, "file_size": 15, "text_off": 60180480, "text_size": 0, "type": "frame"}{"type": "/frame"}
-    # {"file_off": 14508951, "file_size": 74, "text_off": 59792468, "text_size": 9132, "type": "frame"}{"type": "/report"}{"type": "/frame"}
-    # {"file_off": 11295485, "file_size": 60, "text_off": 43934392, "text_size": 5448, "type": "frame"}{"type": "/report"}{"type": "/frame"}
-    # {"file_off": 9654710, "file_size": 28, "text_off": 41709591, "text_size": 1, "type": "frame"}{"type": "/report"}{"type": "/frame"}
-    # {"file_off": 26744767, "file_size": 28, "text_off": 116908344, "text_size": 1, "type": "frame"}{"type": "/report"}{"type": "/frame"}
-    #
-    # TODO: {type: badblob} is not accounted as it's not currently imported during centrifugation stage.
-
-    files = set()
-    use_report = True
-    report_count, datum_count = 0, 0
+def load_autoclaved_index(autoclaved_index):
+    # Returns: {filename: file_sha1 for autoclaved}
+    files = {}
     with gzip.GzipFile(autoclaved_index, 'r') as indexfd:
         for _, doc in autoclaving.stream_json_blobs(indexfd):
+            if 'file"' not in doc:
```

Comment:
I would maybe add a comment above this stating that this is a "fast path" and that the `"` after file is not a typo, but rather a slightly more strict check for the JSON key `"file": something` (also why not `file":`?)

Reply:
Ah ok I see, so you are actually looking for the JSON "value", not the JSON key. I think it's worthwhile to add a comment saying what it is that you are looking for then.